tools/src/main/java/org/apache/kafka/trogdor/workload/ProduceBenchWorker.java

```diff
@@ -88,7 +80,7 @@ public void start(Platform platform, WorkerStatusTracker status,
             throw new IllegalStateException("ProducerBenchWorker is already running.");
         }
         log.info("{}: Activating ProduceBenchWorker with {}", id, spec);
-        this.executor = Executors.newScheduledThreadPool(1,
+        this.executor = Executors.newScheduledThreadPool(2,
```

Comment:
why?

Reply:
We need the second thread so that the StatusUpdater can run in parallel with SendRecords. I will add a comment clarifying this.